backend/src/main.py

```diff
@@ -0,0 +1,58 @@
+
+# General imports =================================================================================
+import time
+import os, sys, git
+import multiprocessing as mp
+
+# Project specific imports ========================================================================
+git_repo = git.Repo(__file__, search_parent_directories=True).git.rev_parse("--show-toplevel")
+sys.path.insert(0, os.path.join(git_repo, "backend", 'proto/Python'))
+sys.path.insert(0, os.path.join(git_repo, "backend"))
+
+import proto.Python.CoreProto_pb2 as ProtoCore
+
+from src.support.CommonLogger import logger
+from src.DatabaseHandler import database_thread
+from src.SerialHandler import SerialDevices as sd, serial_thread
+from src.Utils import RADIO_BAUDRATE, THREAD_MESSAGE_DB_WRITE, UART_BAUDRATE, Utils as utl, WorkQ_Message
+
+
+# Local Procedures ================================================================================
+def initialize_threads():
+        '''
+        Create threads for the backend
+        '''
+        logger.info('Initializing threads')
+        thread_pool = {}
+        uart_workq = mp.Queue()
+        radio_workq = mp.Queue()
+        db_workq = mp.Queue()
+        message_handler_workq = mp.Queue()
+
+        # Create a main thread for handling thread messages
+        thread_pool['message_handler'] = {'thread': None, 'workq': message_handler_workq}
+
+        # Initialize the serial handler threads
+        uart_thread = mp.Process(target=serial_thread, args=('uart', sd.UART, UART_BAUDRATE, uart_workq, message_handler_workq))
+        radio_thread = mp.Process(target=serial_thread, args=('radio', sd.RADIO, RADIO_BAUDRATE, radio_workq, message_handler_workq))
+
+        # Add the threads to the thread pool
+        thread_pool['uart'] = {'thread': uart_thread, 'workq': uart_workq}
+        thread_pool['radio'] = {'thread': radio_thread, 'workq': radio_workq}
+        
+        # Initialize the database handler thread
+        db_thread = mp.Process(target=database_thread, args=('database', db_workq, message_handler_workq))
+        # Add the database thread to the thread pool
+        thread_pool['database'] = {'thread': db_thread, 'workq': db_workq}
+        
+        utl.thread_pool = thread_pool
+        return
+
+if __name__ == "__main__":
+  utl()
```

Comment:
Maybe add a quick argparse, or just a variable we can change here so we can change the logging level. You can implement this as a function in CommonLogger for example called `setup_logger(level)` which runs `logger.add()` with a different log level based on the one that's given.

We could also potentially add a function like `change_log_level()` that just removes and adds back the logger with a different level, but we probs wouldn't use this

Reply:
I like this idea andd think it should be added or something like it, however there is a problem with this since we are using mp. I realized that if something isn't staticly allocated the threads will use whatever the static value is.

For example I was trying to dynamically turn off an on logging to terminal from main (which would then go into common logger and either turn off or on logging to terminal) but then when we create the othere threads they do not look at what the main thread had already done to the logger andd just use the static version which was default connected to terminal and default dont add to the log fille.

A way that this could actually be fixed is by adding that backend handler thread which can work on making sure the logs are all handled in on place... maybe I'm worried about it getting overeloaded with too many logs but when we are out of the development portion of the backend we can remove a ton of the super super verbose logs maybe

TLDR: I think we should add this but not in this commit I'll make another todo in the readme and start the overhall after this merge.

Reply:
sounds good, another idea that may work good for this is to just have a Configuration.py with global constants such as that DISABLE_LOGGING and log level cause that way we _should_ only ever have to touch one file to change how it's workin but yea i'm good with doing it after merge

Reply:
yeah that would be an easier fix plus can work for other constants like uart buads, ports, db url etc..

Reply:
I tested it with the blocking and seems to work fine

Reply:
oh i diddnt do this but i addded a todo for the logs stuff